include/device/map.h

```diff
@@ -38,17 +46,16 @@ static inline int find_mapid_by_addr(IOMap *maps, int size, paddr_t addr) {
   int i;
   for (i = 0; i < size; i ++) {
     if (map_inside(maps + i, addr)) {
-      difftest_skip_ref();
       return i;
     }
   }
   return -1;
 }
 
 void add_pio_map(const char *name, ioaddr_t addr,
```

Comment:
I would prefer adding another function called `add_mmio_map_with_diff` with this `mmio_diff_flag`. Then this existing function does not need to be changed and existing code is also unchanged. Only these devices that we want diff should call the new function with the appropriate flags.

Reply:
good idea, i'll do it

Reply:
I added the `add_mmio_map_with_diff` api